### ISSUE TITLE
Method to get the bytesize before serialization?

### ISSUE THREAD

**Issue by didactic-drunk** [jeromegn/protobuf.cr#31]
The google implementation has one.

[https://developers.google.com/protocol-buffers/docs/reference/cpp/google.protobuf.message_lite#MessageLite.ByteSizeLong.details](https://developers.google.com/protocol-buffers/docs/reference/cpp/google.protobuf.message_lite#MessageLite.ByteSizeLong.details)

**Comment by jeromegn** [comment]
Sounds useful

**Comment by jgaskins** [comment]
Agreed, this would be [useful for gRPC](https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md#requests). For the gRPC library I'm working on (speaking of which, I really should push it to GitHub), I've just been serializing the message to prefix it with the length but having a `Protobuf::Message#bytesize` method would clean that code up a lot.
